source/common/http/utility.cc

```diff
@@ -121,6 +121,41 @@ std::string Utility::parseCookieValue(const HeaderMap& headers, const std::strin
   return state.ret_;
 }
 
+bool Utility::hasSetCookie(const HeaderMap& headers, const std::string& key) {
+
+  struct State {
+    std::string key_;
+    bool ret_;
+  };
+
+  State state;
+  state.key_ = key;
+  state.ret_ = false;
+
+  headers.iterate(
```

Comment:
This is basically the same as parseCookieValue but skipping the string copy, right?  I wonder if it's worth having a comon helper with a boolean to copy the string or not.  Optional - if you prefer it this way feel free to leave it this way

Reply:
+1

Reply:
The Cookie header has multiple cookie values inline whereas a Set-Cookie header has only a single cookie value. There's an extra inner loop in parseCookieValue that hasSetCookie doesn't need.